### PR TITLE
Add option to flatten JSON with FULL_MESSAGE_IN_JSON_EXPAND_VALUE format

### PR DESCRIPTION
**Pull request by cbornet**
### Motivation
There are use-cases where it's beneficial to flatten the JSON before pushing to Kinesis so it can be easily loaded into data lakes / data warehouses. See more details here: https://aws.amazon.com/blogs/big-data/simplify-querying-nested-json-with-the-aws-glue-relationalize-transform/ .

### Modifications
Add a setting `jsonFlatten ` : When set to true and the message format is FULL_MESSAGE_IN_JSON_EXPAND_VALUE the output JSON will be flattened.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change added tests and can be verified as follows:
* Run `org.apache.pulsar.io.kinesis.UtilsTest:: testKeyValueSerializeRecordToJsonExpandingValue`

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no) yes : `com.github.wnameless.json:json-flattener` in `pulsar-io-kinesis`
  - The public API: (yes / no) no
  - The schema: (yes / no / don't know) no
  - The default values of configurations: (yes / no) no
  - The wire protocol: (yes / no) no
  - The rest endpoints: (yes / no) no
  - The admin cli options: (yes / no) no
  - Anything that affects deployment: (yes / no / don't know) no

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [x] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by nicoloboschi — APPROVED** [review]
cool! LGTM

**Review by lhotari — APPROVED** [review]
Great work

**Review by dlg99 — APPROVED** [review]
LGTM

**Comment by dlg99** [comment]
@fantapsody @RobertIndie Please take a look too

**Review by fantapsody — APPROVED** [review]

File: pulsar-io/kinesis/pom.xml [hunk 1]
```diff
@@ -84,6 +84,12 @@
       <version>${avro.version}</version>
     </dependency>
 
+    <dependency>
+      <groupId>com.github.wnameless.json</groupId>
+      <artifactId>json-flattener</artifactId>
+      <version>0.13.0</version>
```
Comment (Shoothzj) [inline pulsar-io/kinesis/pom.xml:90]: it's better to define a version instead of hardcode version here. like avro

**Comment by Anonymitaet** [comment]
@momo-jun a soft reminder: here is a PR w/ doc-required label

**Comment by momo-jun** [comment]
@Anonymitaet The doc changes have been included in this PR.

**Comment by Anonymitaet** [comment]
@momo-jun could you please help review the doc updates? Thanks

**Review by RobertIndie — APPROVED** [review]

File: site2/docs/io-kinesis-sink.md [hunk 2]
```diff
@@ -12,9 +12,11 @@ The configuration of the Kinesis sink connector has the following property.
...
+`jsonIncludeNonNulls`|boolean|false|true|Value that indicates that only properties with non-null values are to be included when using MessageFormat.FULL_MESSAGE_IN_JSON_EXPAND_VALUE.
+`jsonFlatten`|boolean|false|false|When set to true and the message format is FULL_MESSAGE_IN_JSON_EXPAND_VALUE the output JSON will be flattened.
```
Comment (momo-jun) [inline site2/docs/io-kinesis-sink.md:19]:
```suggestion
`jsonFlatten`|boolean|false|false|When it is set to `true` and the message format is `FULL_MESSAGE_IN_JSON_EXPAND_VALUE`, the output JSON is flattened.
```

File: site2/docs/io-kinesis-sink.md [hunk 3]
```diff
@@ -12,9 +12,11 @@ The configuration of the Kinesis sink connector has the following property.
...
 |------|----------|----------|---------|-------------|
 `messageFormat`|MessageFormat|true|ONLY_RAW_PAYLOAD|Message format in which Kinesis sink converts Pulsar messages and publishes to Kinesis streams.<br/><br/>Below are the available options:<br/><br/><li>`ONLY_RAW_PAYLOAD`: Kinesis sink directly publishes Pulsar message payload as a message into the configured Kinesis stream. <br/><br/><li>`FULL_MESSAGE_IN_JSON`: Kinesis sink creates a JSON payload with Pulsar message payload, properties and encryptionCtx, and publishes JSON payload into the configured Kinesis stream.<br/><br/><li>`FULL_MESSAGE_IN_FB`: Kinesis sink creates a flatbuffer serialized payload with Pulsar message payload, properties and encryptionCtx, and publishes flatbuffer payload into the configured Kinesis stream.<br/><br/><li>`FULL_MESSAGE_IN_JSON_EXPAND_VALUE`: Kinesis sink sends a JSON structure containing the record topic name, key, payload, properties and event time. The record schema is used to convert the value to JSON.
+`jsonIncludeNonNulls`|boolean|false|true|Value that indicates that only properties with non-null values are to be included when using MessageFormat.FULL_MESSAGE_IN_JSON_EXPAND_VALUE.
```
Comment (momo-jun) [inline site2/docs/io-kinesis-sink.md:18]:
```suggestion
`jsonIncludeNonNulls`|boolean|false|true|Only the properties with non-null values are included when the message format is `FULL_MESSAGE_IN_JSON_EXPAND_VALUE`.
```

**Review by momo-jun — COMMENTED** [review]
Hi @cbornet, thanks for contributing to docs!
I proposed minor changes to the description to make it more readable.

File: pulsar-io/kinesis/pom.xml (hunk 1 above)
Comment (cbornet) [reply pulsar-io/kinesis/pom.xml:90]: A lot of deps in this pom don't use versions in properties (eg. amazon-kinesis-client)... but ok...

**Review by momo-jun — APPROVED** [review]
Doc review completed. LGTM. 
@Anonymitaet

**Comment by Anonymitaet** [comment]
@momo-jun thanks!

File: pulsar-io/kinesis/pom.xml (hunk 1 above)
Comment (cbornet) [reply pulsar-io/kinesis/pom.xml]: Done.
@Shoothzj can you give another look ?

**Pull request merged**